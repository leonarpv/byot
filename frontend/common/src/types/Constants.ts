import logoMetadata from '../static/meta/full-svg.json';

export class Constants {
  public static readonly LOGO_SIZE: {
    height: number;
    width: number;
    type: string;
  } = logoMetadata;
}
