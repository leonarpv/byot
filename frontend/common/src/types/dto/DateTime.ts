import {IDateTime} from '../interfaces/IDateTime';
import moment from 'moment';

export class DateTime implements IDateTime {
  public humanReadable: string;
  public iso: string;

  constructor(dateTime?: IDateTime) {
    if (!dateTime) {
      const curr = moment();
      this.iso = curr.toISOString();
      this.humanReadable = curr.format('DD.MM.YYYY HH:ss');
    } else {
      this.iso = dateTime.iso;
      this.humanReadable = dateTime.humanReadable;
    }
  }
}
