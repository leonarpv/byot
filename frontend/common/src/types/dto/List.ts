import {IListMeta} from '../interfaces/IListMeta';
import {IList, ListEntity} from '../interfaces/IList';

export class List<T extends ListEntity> implements IList<T> {
  public entries: T[];
  public meta: IListMeta;
  constructor(list?: IList<T>);
  constructor(entries: T[], meta: IListMeta);
  constructor(entries?: T[] | IList<T>, meta?: IListMeta) {
    if (Array.isArray(entries)) {
      this.entries = entries;
      this.meta = meta!;
    } else {
      this.entries = entries?.entries || [];
      this.meta = entries?.meta || {totalCount: 0};
    }
  }
}
