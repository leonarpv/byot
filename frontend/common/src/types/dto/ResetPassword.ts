import {IResetPassword} from '../interfaces/IResetPassword';

export class ResetPassword implements IResetPassword {
  public newPassword: string;
  public passwordRepeat: string;
  public token: string;
  constructor({newPassword = '', passwordRepeat = '', token = ''}: Partial<IResetPassword> = {}) {
    this.newPassword = newPassword;
    this.passwordRepeat = passwordRepeat;
    this.token = token;
  }
}
