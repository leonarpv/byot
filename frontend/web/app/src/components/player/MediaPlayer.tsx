import * as React from 'react';
import {MediaPlayerContainer} from './MediaPlayerContainer';
import {IMedia} from '@byot-frontend/common/src/types/interfaces/IMedia';
import {SourceType} from '@byot-frontend/common/src/types/enums/SourceType';
import {YoutubeMediaPlayer} from './providers/YoutubeMediaPlayer';

interface Props {
  media: IMedia;
}

export const MediaPlayer: React.FC<Props> = (props: Props) => {
  return (
    <MediaPlayerContainer>
      {props.media.source.sourceType === SourceType.YOUTUBE && <YoutubeMediaPlayer media={props.media} />}
    </MediaPlayerContainer>
  );
};
