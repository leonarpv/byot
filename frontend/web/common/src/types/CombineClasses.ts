import {ClassKeyOfStyles, ClassNameMap} from '@material-ui/styles/withStyles';

export type ClassesObject<C> = {classes?: Partial<ClassNameMap<ClassKeyOfStyles<C>>>};

export type CombineClasses<
  T1 extends ClassesObject<C1>,
  T2 extends ClassesObject<C2>,
  C1 = any,
  C2 = any
> = Omit<T1, 'classes'> &
  Omit<T2, 'classes'> & {
    classes?: Partial<T1['classes'] & T2['classes']>;
  };
