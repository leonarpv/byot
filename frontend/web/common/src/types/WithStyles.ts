import * as React from 'react';
import {Theme} from '@material-ui/core';
import {ClassNameMap, Styles} from '@material-ui/core/styles/withStyles';
import {ClassKeyOfStyles} from '@material-ui/styles/withStyles';

export interface WithTheme {
  theme?: Theme;
  innerRef?: React.Ref<any> | React.RefObject<any>;
}

export type WithStyles<
  StylesOrClassKey extends string | Styles<any, any, any> = string,
  IncludeTheme extends boolean | undefined = false
> = (IncludeTheme extends true ? {theme: Theme} : {}) & {
  classes?: Partial<ClassNameMap<ClassKeyOfStyles<StylesOrClassKey>>>;
  className?: string;
  theme?: Theme;
  style?: React.CSSProperties;
};
