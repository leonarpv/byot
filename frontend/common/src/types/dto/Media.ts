import {IMedia} from '../interfaces/IMedia';
import {ISource} from '../interfaces/ISource';
import {IDateTime} from '../interfaces/IDateTime';
import {DateTime} from './DateTime';
import {Source} from './Source';

export class Media implements IMedia {
  public id: string;
  public label: string;
  public source: ISource;
  public updatedAt: IDateTime;
  public createdAt?: IDateTime;

  constructor({id, label, source, updatedAt, createdAt}: IMedia) {
    this.createdAt = createdAt instanceof DateTime ? createdAt : new DateTime(createdAt);
    this.updatedAt = updatedAt && updatedAt instanceof DateTime ? updatedAt : new DateTime(updatedAt);
    this.id = id;
    this.label = label;
    this.source = source instanceof Source ? source : new Source(source);
  }
}
