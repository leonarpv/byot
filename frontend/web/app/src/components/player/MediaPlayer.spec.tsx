import * as React from 'react';
import {render} from '@testing-library/react';
import {MediaPlayer} from './MediaPlayer';
import {media} from '@byot-frontend/common/test/fixtures/dto/Media';
import {SourceType} from '../../../../../../common/graphql/ts/types';

jest.mock('./providers/YoutubeMediaPlayer', () => ({
  __esModule: true,
  YoutubeMediaPlayer: () => <div data-testid="youtube-media-player" />,
}));
jest.mock('./MediaPlayerContainer', () => ({
  __esModule: true,
  MediaPlayerContainer: ({children}: any) => <div>{children}</div>,
}));

describe('<MediaPlayer/>', () => {
  it('should render with youtube provider', () => {
    const m = media();
    m.source.sourceType = SourceType.YOUTUBE;
    const {container} = render(<MediaPlayer media={m} />);
    expect(container).toMatchSnapshot();
  });
});
