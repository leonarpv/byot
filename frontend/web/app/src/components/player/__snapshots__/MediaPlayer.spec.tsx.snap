// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaPlayer/> should render with youtube provider 1`] = `
<div>
  <div>
    <div
      data-testid="youtube-media-player"
    />
  </div>
</div>
`;
